pkg/web/static/js/userProfile.js

```diff
@@ -0,0 +1,53 @@
+// Helper function to add the same event handler for multiple events
+const addEventListeners = (el, evts, fn) => {
+  evts.split(' ').forEach(e => el.addEventListener(e, fn, false));
+}
+
+// Elements referenced by this script
+const changePwdModal = document.getElementById('change_pwd_modal');
+const changePwdForm = document.getElementById('change-pwd-form')
+const closePwdModalBtn = document.getElementById('close-pwd-mdl-btn');
+const passwordInput = document.getElementById('password');
+const confirmPassword = document.getElementById('confirm-password');
+const changePasswordBtn = document.getElementById('change-password-btn');
+
+// Reset the form when change password modal is closed in case user entered some values without submitting.
+closePwdModalBtn.addEventListener('click', () => {
+  changePwdForm.reset();
+});
+
+// Enable the change password button when the password and confirm password inputs are identical.
+// TODO: should we add any other visual indication that the passwords do not match?
+const checkPasswords = e => {
+  if (passwordInput.value == confirmPassword.value && passwordInput.value != '') {
+    changePasswordBtn.removeAttribute('disabled');
+  } else {
+    changePasswordBtn.setAttribute('disabled', 'true');
+  }
+}
+
+// TODO: is this a complete list of events that should be used to detect if the passwords match?
+addEventListeners(passwordInput, 'change keyup paste cut', checkPasswords);
+addEventListeners(confirmPassword, 'change keyup paste cut', checkPasswords);
+
+// Write code to run after an htmx request event.
+// TODO: should we do a pre-flight check as a last sanity check that the passwords match?
```

Comment:
@daniellemaxwell I just wanted to call your attention to these TODOs I put in during my review -- let me know what you think about these questions.

Reply:
`setCustomValidity` could be used to provide specific wording to users here: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setCustomValidity

I noticed that the button doesn't return to the disabled state if a user enters a password that doesn't meet the validation requirements. I'll make an update.

Reply:
oh, setCustomValidity is awesome - so much better than putting class state on the inputs!

Reply:
I was going to use `setCustomValidity` but the message didn't appear when I clicked the submit button. I must've missed something, so I'll create a story to revisit and make the update.

Reply:
Cheers!